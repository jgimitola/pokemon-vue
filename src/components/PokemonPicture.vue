<script>
export default {
  name: "PokemonPicture",
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      default: false,
      required: true,
    },
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
    },
  },
};
</script>

<template>
  <div class="pokemon-container">
    <img v-if="showPokemon" class="fade-in" :src="imgSrc" alt="pokemon" />
    <img v-else class="hidden-pokemon" :src="imgSrc" alt="pokemon" />
  </div>
</template>

<style scoped>
.pokemon-container {
  height: 200px;
  display: flex;
  justify-content: center;
}

img {
  height: 200px;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>
