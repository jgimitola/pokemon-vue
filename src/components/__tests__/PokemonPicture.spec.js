import { shallowMount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import PokemonPicture from "@/components/PokemonPicture.vue";

describe("PokemonPicture", () => {
  test("Match snapshot", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: 1, showPokemon: false },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Show hidden image with pokemon id = 100", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: 100, showPokemon: false },
    });

    const imgs = wrapper.findAll("img");

    expect(imgs.length).toBe(1);
    expect(imgs[0].classes("hidden-pokemon")).toBe(true);
  });

  test("Show answer image", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: { pokemonId: 100, showPokemon: true },
    });

    const imgs = wrapper.findAll("img");

    expect(imgs.length).toBe(1);
    expect(imgs[0].classes("hidden-pokemon")).toBe(false);
    expect(imgs[0].classes("fade-in")).toBe(true);
    expect(imgs[0].attributes("src")).toBe(
      `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${100}.svg`
    );
  });
});
