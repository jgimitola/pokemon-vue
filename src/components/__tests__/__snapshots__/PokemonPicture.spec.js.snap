// Vitest Snapshot v1

exports[`PokemonPicture > Match snapshot 1`] = `"<div data-v-34cacee9=\\"\\" class=\\"pokemon-container\\"><img data-v-34cacee9=\\"\\" class=\\"hidden-pokemon\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg\\" alt=\\"pokemon\\"></div>"`;
